### PR TITLE
Tracing fixes for POSIX arch boards

### LINKED ISSUE #13357: Tracing hooks problem on POSIX
**Describe the bug**
Tracing hook do not work properly on POSIX (native) architecture.

**To Reproduce**
Steps to reproduce (using philosophers example):
1. mkdir build; cd build
2. cmake .. -DBOARD=native_posix
3. In menuconfig add "Tracing via Common Trace Format Support"
4. make run
5. See the log (for example with babeltrace)

**Expected behavior**
Thread swtiching in and out should match (we shouldn't have multiple logs from thread switching in in a row without thread switching out). Similar problem occurs for isr_enter (there is no matching isr_exit).

**Impact**
Current tracing hooks are not going to work properly for debug purposes on POSIX architecture.

**Screenshots or console output**
```
[01:00:00.018720000] (+0.000000000) 0 thread_pending: { thread_id = 134560504 }
[01:00:00.018720000] (+0.000000000) 0 thread_switched_out: { thread_id = 134560504 }
[01:00:00.018720000] (+0.000000000) 0 thread_switched_in: { thread_id = 134560504 }
[01:00:00.018720000] (+0.000000000) 0 thread_switched_in: { thread_id = 134560416 }
[01:00:00.018720000] (+0.000000000) 0 thread_switched_out: { thread_id = 134560416 }
[01:00:00.018720000] (+0.000000000) 0 thread_switched_in: { thread_id = 134560416 }
[01:00:00.018720000] (+0.000000000) 0 thread_switched_in: { thread_id = 134562080 }
[01:00:00.018720000] (+0.000000000) 0 idle:
[01:00:00.018730000] (+0.000010000) 0 isr_enter:
[01:00:00.018730000] (+0.000000000) 0 idle:
[01:00:00.018740000] (+0.000010000) 0 isr_enter:
[01:00:00.018740000] (+0.000000000) 0 idle:
[01:00:00.018750000] (+0.000010000) 0 isr_enter:
[01:00:00.018750000] (+0.000000000) 0 idle:
[01:00:00.018760000] (+0.000010000) 0 isr_enter:
[01:00:00.018760000] (+0.000000000) 0 idle:
```

**Environment:**
 - OS: Ubuntu 16.04 LTS
 - Toolchain: Zephyr SDK 0.9.5

### PR DESCRIPTION
**Pull request by aescolar**
Fix the tracing calls for the posix arch:
* isr_exit was missing
* z_sys_trace_thread_switched_in() was called where it shouldn't
* as a bonus trace the switch to the main thread

Fixes #13357

### CODE REVIEW

**Comment by aescolar** [comment]
CC: @MarekPieta

**Review by nashif — APPROVED** [review]

**Review by mped-oticon — APPROVED** [review]
LGTM

**Comment by aescolar** [comment]
Note: The coverage changes in the nrf52 part is due to the change in the  nrf52_bsim board triggering more tests being run in that board in this run. Only the boards/posix/*/irq_handler.c changes are "real".

File: arch/posix/core/swap.c [hunk 1]
```diff
@@ -93,7 +93,14 @@ void _arch_switch_to_main_thread(struct k_thread *main_thread,
 	posix_thread_status_t *ready_thread_ptr =
 			(posix_thread_status_t *)
 			_kernel.ready_q.cache->callee_saved.thread_status;
+
+#ifdef CONFIG_TRACING
```
Comment (MarekPieta) [inline arch/posix/core/swap.c]: Are ifdefs here necessary?

Comment (aescolar) [reply arch/posix/core/swap.c]: Not really :)

**Pull request merged**